sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -1256,6 +1273,25 @@ case class LocalLimit(limitExpr: Expression, child: LogicalPlan) extends OrderPr
     copy(child = newChild)
 }
 
+/**
+ * A global (coordinated) limit with offset. This operator can skip at most `offsetExpr` number and
+ * emit at most `limitExpr` number in total.
```

Comment:
Similar to above, can we mention in this comment:

1. Whether an accompanying LIMIT is also required, or the OFFSET is allowed to stand alone
2. The semantics of which rows get skipped or preserved in the presence of an accompanying LIMIT. For example, if we have LIMIT 10 OFFSET 5, do we (a) impose a total limit of 10 + 5 = 15 rows and then discard the first 5, leaving 10 rows remaining?
3. What are the boundary conditions, e.g. min and max allowed values?
4. What is the difference between this and a chain of regular Limit + Offset operators?

Reply:
1. The constructor do the limit.
2. OK.
3. The boundary conditions is also mentioned in `CheckAnalysis.scala`.
4. `Offset` is a logical representation and `GlobalLimitAndOffset` more closer physical plan.